src/modules/cpu/kernel/normalize_generic.hpp

```diff
@@ -0,0 +1,513 @@
+    }
+}
+
+void normalize_3D_tensor_axis1_toggle(Rpp32f *srcPtr, RpptGenericDescPtr srcGenericDescPtr, Rpp32f *dstPtr, RpptGenericDescPtr dstGenericDescPtr,
+                         Rpp32f *meanPtr, Rpp32f *stdDevPtr, Rpp32f scale, Rpp32f shift, Rpp32u *paramStride)
+{
+    Rpp32f *srcPtrTemp = srcPtr;
+    Rpp32f *dstPtrTemp[srcGenericDescPtr->dims[3]];
+    dstPtrTemp[0] = dstPtr;
+    for(int i = 1; i < srcGenericDescPtr->dims[3]; i++)
+        dstPtrTemp[i] = dstPtrTemp[i-1] + dstGenericDescPtr->strides[1];
+
+    Rpp32s paramIdx = 0;
+    Rpp32f multiplier[srcGenericDescPtr->dims[3]];
+
+    for(int i = 0; i < srcGenericDescPtr->dims[3]; i++)
+        multiplier[i] = scale / stdDevPtr[i];
+
+    for(Rpp32u i = 0; i < srcGenericDescPtr->dims[1]; i++)
+    {
+        Rpp32f *srcPtrRow = srcPtrTemp;
+        Rpp32f *dstPtrRow[srcGenericDescPtr->dims[3]];
+        for(int l = 0; l < srcGenericDescPtr->dims[3]; l++)
+            dstPtrRow[l] = dstPtrTemp[l];
+        for(Rpp32u j = 0; j < srcGenericDescPtr->dims[2]; j++)
+        {
+            Rpp32f *srcPtrRowTemp = srcPtrRow;
+            Rpp32f *dstPtrRowTemp[srcGenericDescPtr->dims[3]];
+            for(int l = 0; l < srcGenericDescPtr->dims[3]; l++)
+                dstPtrRowTemp[l] = dstPtrRow[l];
+            for(Rpp32u k = 0; k < srcGenericDescPtr->dims[3]; k++)
+            {
+                *dstPtrRowTemp[k]++ = ((*srcPtrRowTemp++ - meanPtr[paramIdx]) * multiplier[paramIdx]) + shift;
+                paramIdx += paramStride[1];
+            }
+            paramIdx = (!paramStride[2]) ? 0 : paramIdx + paramStride[2];
+            srcPtrRow += srcGenericDescPtr->strides[2];
+            for(int l = 0; l < srcGenericDescPtr->dims[3]; l++)
+                dstPtrRow[l] += dstGenericDescPtr->strides[3];
+        }
+        srcPtrTemp += srcGenericDescPtr->strides[1];
+        for(int l = 0; l < srcGenericDescPtr->dims[3]; l++)
+            dstPtrTemp[l] += dstGenericDescPtr->strides[2];
+    }
+}
+
+void normalize_3D_tensor_avx_axis1(Rpp32f *srcPtr, RpptGenericDescPtr srcGenericDescPtr, Rpp32f *dstPtr, RpptGenericDescPtr dstGenericDescPtr,
+                                   Rpp32f *meanPtr, Rpp32f *stdDevPtr, Rpp32f scale, Rpp32f shift, Rpp32u *paramStride, Rpp32u bufferLength)
+{
+    Rpp32s paramIdx = 0;
+    Rpp32u alignedLength = (bufferLength / 16) * 16;
+    Rpp32u OuterDim = srcGenericDescPtr->dims[1];
+
+    // set shift, mean and stddev
+    __m256 pShift = _mm256_set1_ps(shift);
+    __m256 pScale = _mm256_set1_ps(scale);
+    __m256 pMean1 = _mm256_loadu_ps(meanPtr);
+    __m256 pMean2 = _mm256_loadu_ps(meanPtr + 8);
+    __m256 pStdDev1 = _mm256_loadu_ps(stdDevPtr);
+    __m256 pStdDev2 = _mm256_loadu_ps(stdDevPtr + 8);
+    __m256 pMultiplier1 = _mm256_div_ps(pScale, pStdDev1); // Using division operation as stddev is a vector and scale is a scalar thus can't be pre-divided
+    __m256 pMultiplier2 = _mm256_div_ps(pScale, pStdDev2);
```

Comment:
If all the values from L263 - L270 are constant
They can be set only once in a batch and can be passed as a input to this function, instead of setting the same value for every sample in a batch